tests/yaml-test-suite.rs

```diff
@@ -34,15 +34,26 @@ fn main() -> Result<()> {
 
 fn run_yaml_test(test: &Test<YamlTest>) -> Outcome {
     let desc = &test.data;
-    let actual_events = parse_to_events(&desc.yaml);
-    let events_diff = actual_events.map(|events| events_differ(&events, &desc.expected_events));
+    let reporter = parse_to_events(&desc.yaml);
+    let actual_events = reporter.as_ref().map(|reporter| &reporter.events);
+    let events_diff = actual_events.map(|events| events_differ(events, &desc.expected_events));
     let mut error_text = match (&events_diff, desc.expected_error) {
         (Ok(x), true) => Some(format!("no error when expected: {x:#?}")),
         (Err(_), true) | (Ok(None), false) => None,
         (Err(e), false) => Some(format!("unexpected error {e:?}")),
         (Ok(Some(diff)), false) => Some(format!("events differ: {diff}")),
     };
 
+    if let Some(span_error) = reporter
+        .as_ref()
+        .ok()
+        .and_then(|reporter| reporter.span_failures.first())
+    {
+        return Outcome::Failed {
+            msg: Some(span_error.clone()),
+        };
```

Comment:
I'd like a couple changes to this:
- Check the errors in the spans _after_ having output a potential YAML parsing error. As it is now, if there is an inconsistency in the spans and the test from the YAML test suite fails, this `return` short circuits the output in the `if` below.
- Extract the `if` below to a separate function (taking notably the error message and marker as parameter) so that both YAML test suite errors and span inconsistencies benefit from a context output to aid in debugging.

If you don't have the time, just let me know and I'll do it. I appreciate your contributions <3